File: src/main/java/com/github/wgzhao/dbquery/dto/DatabaseType.java
package com.github.wgzhao.dbquery.dto;

import lombok.Getter;

@Getter
public enum DatabaseType {
    MYSQL("mysql"),
    POSTGRESQL("postgresql"),
    SQLSERVER("sqlserver"),
    ORACLE("oracle"),
    DB2("db2"),
    H2("h2"),
    HSQLDB("hsqldb"),
    SQLITE("sqlite"),
    MARIADB("mariadb"),
    OTHER("other");

    private final String value;

    DatabaseType(String value) {
        this.value = value;
    }

    public static DatabaseType fromValue(String value) {
        for (DatabaseType type : DatabaseType.values()) {
            if (type.getValue().equalsIgnoreCase(value)) {
                return type;
            }
        }
        return OTHER;
    }
}

File: src/main/kotlin/com/github/wgzhao/dbquery/dto/DatabaseType.kt
package com.github.wgzhao.dbquery.dto

enum class DatabaseType(private val value: String) {
    MYSQL("mysql"),
    POSTGRESQL("postgresql"),
    SQLSERVER("sqlserver"),
    ORACLE("oracle"),
    DB2("db2"),
    H2("h2"),
    HSQLDB("hsqldb"),
    SQLITE("sqlite"),
    MARIADB("mariadb"),
    OTHER("other");

    companion object {
        fun fromValue(value: String?): DatabaseType {
            for (type in DatabaseType.entries) {
                if (type.value.equals(value, ignoreCase = true)) {
                    return type
                }
            }
            return DatabaseType.OTHER
        }
    }
}
